### PR TITLE
Fix device plugin node ready test assertion

### LINKED ISSUE #126915: Failing test: `[It] [sig-node] Device Plugin [NodeFeature:DevicePlugin] [Serial] DevicePlugin [Serial] [Disruptive] Keeps device plugin assignments across node reboots (no pod restart, no device plugin re-registration) [Flaky]`
### Which jobs are failing?

ci-crio-cgroupv1-node-e2e-flaky

### Which tests are failing?

[It] [sig-node] Device Plugin [NodeFeature:DevicePlugin] [Serial] DevicePlugin [Serial] [Disruptive] Keeps device plugin assignments across node reboots (no pod restart, no device plugin re-registration) [Flaky]

### Since when has it been failing?

Since d770dd695

### Testgrid link

https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv1-node-e2e-flaky

### Reason for failure (if possible)

Under evaluation.

### Anything else we need to know?

cc @kubernetes/sig-node-cri-o-test-maintainers 

### Relevant SIG(s)

/sig node

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:
Introduced in d770dd695a4ef15b9fb41e3ef43135f7df55a882 and high likely the issue caused in the failing test:
https://github.com/kubernetes/kubernetes/issues/126915

#### Which issue(s) this PR fixes:

Fixes #126915

#### Special notes for your reviewer:
/hold

for verification
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Review by pacoxu — APPROVED** [review]
https://github.com/kubernetes/kubernetes/pull/121888/files#diff-74c81c82cb802fc4acfc5daddeaf9d17f040b609ef2325c0134eb113d77235a8R717-R723 is not valid.

/lgtm

**Comment by saschagrunert** [comment]
Yeah I think we cannot test it before merge because it looks like the only suite running that test is: https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv1-node-e2e-flaky

/unhold

File: test/e2e_node/device_plugin_test.go [hunk 1]
```diff
@@ -803,13 +803,11 @@ func testDevicePluginNodeReboot(f *framework.Framework, pluginSockDir string) {
...
-				if err != nil {
-					return err
-				}
-				return nil
-			}, time.Minute, time.Second).Should(gomega.Equal(1), "one node should be ready")
+			gomega.Eventually(ctx, func(ctx context.Context) bool {
+				nodes, err := e2enode.TotalReady(ctx, f.ClientSet)
+				framework.ExpectNoError(err)
+				return nodes == 1
+			}, time.Minute, time.Second).Should(gomega.BeTrueBecause("node should be ready"))
```
Comment (pohly) [inline test/e2e_node/device_plugin_test.go:810]: Don't make assertions inside a `gomega.Eventually`. They interact poorly with each other (test is marked as failed immediately, polling may or may not continue).

Instead, return the count and the error, then use `gomega.Equal(1)`.

Comment (pohly) [reply test/e2e_node/device_plugin_test.go:810]: /cancel lgtm

Comment (saschagrunert) [reply test/e2e_node/device_plugin_test.go]: Done.

File: test/e2e_node/device_plugin_test.go [hunk 2]
```diff
@@ -803,13 +803,10 @@ func testDevicePluginNodeReboot(f *framework.Framework, pluginSockDir string) {
...
 		ginkgo.BeforeEach(func(ctx context.Context) {
 			ginkgo.By("Wait for node to be ready")
-			gomega.Eventually(ctx, func(ctx context.Context) error {
-				_, err := e2enode.TotalReady(ctx, f.ClientSet)
-				if err != nil {
-					return err
-				}
-				return nil
-			}, time.Minute, time.Second).Should(gomega.Equal(1), "one node should be ready")
+			gomega.Eventually(ctx, e2enode.TotalReady).
```
Comment (ffromani) [inline test/e2e_node/device_plugin_test.go:806]: `e2enode.TotalReady` returns two value. Does `gomega.BeEquivalentTo` automagically handles the `err` part?

Comment (saschagrunert) [reply test/e2e_node/device_plugin_test.go:806]: It only checks the first return value, I assume it’s fine to skip err, other e2e tests do the same.

Comment (pohly) [reply test/e2e_node/device_plugin_test.go:806]: The error is checked by `gomega.Eventually`. The matcher only gets invoked when the error is nil.

/lgtm

Comment (saschagrunert) [reply test/e2e_node/device_plugin_test.go:806]: Good to know, thanks!

**Comment by pohly** [comment]
As SIG Testing lead I am also going to approve because I was involved in the initial change which led to this.

/approve

**Pull request merged**